src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/partition.py

```diff
@@ -10,39 +10,47 @@
 """
 
 from boto3.session import Session
+from botocore.exceptions import UnknownRegionError
 
-COMPATIBLE_PARTITIONS = ['aws-us-gov', 'aws']
 
+class IncompatibleRegionError(Exception):
+    """Raised in case the regions is not supported."""
 
-class IncompatiblePartitionError(Exception):
-    """Raised in case the partition is not supported."""
 
 
 def get_partition(region_name: str) -> str:
     """Given the region, this function will return the appropriate partition.
 
-    :param region_name: The name of the region (us-east-1, us-gov-west-1)
+    :param region_name: The name of the region (us-east-1, us-gov-west-1, cn-north-1)
+    :raises IncompatibleRegionError: If the provided region is not supported.
     :return: Returns the partition name as a string.
     """
-    partition = Session().get_partition_for_region(region_name)
-    if partition not in COMPATIBLE_PARTITIONS:
-        raise IncompatiblePartitionError(
-            f'The {partition} partition is not supported by this version of '
-            'ADF yet.'
-        )
-
+    try:
+        partition = Session().get_partition_for_region(region_name)
+    except UnknownRegionError as e:
+        raise IncompatibleRegionError(f"The region {region_name} is not supported.") from e
     return partition
 
 
 def get_organization_api_region(region_name: str) -> str:
     """
     Given the current region, it will determine the partition and use
-    that to return the Organizations API region (us-east-1 or us-gov-west-1)
+    that to return the Organizations API region (us-east-1 or us-gov-west-1 or cn-northwest-1)
 
-    :param region_name: The name of the region (eu-west-1, us-gov-east-1)
+    :param region_name: The name of the region (eu-west-1, us-gov-east-1 or cn-northwest-1)
     :return: Returns the AWS Organizations API region to use as a string.
     """
-    if get_partition(region_name) == 'aws-us-gov':
-        return 'us-gov-west-1'
+    if get_partition(region_name) == "aws-us-gov":
+        return "us-gov-west-1"
+    if get_partition(region_name) == "aws-cn":
+        return "cn-northwest-1"
+    return "us-east-1"
+
 
-    return 'us-east-1'
+def get_aws_domain(region_name: str) -> str:
+    """
+    Get AWS domain suffix
+    """
+    if region_name.startswith("cn-north"):
+        return f"amazonaws.com.{region_name.split('-')[0]}"
```

Comment:
If the region_name starts with cn-, the `region_name.split('-')[0]` would always be `cn`.
Hence we can simplify this to:

```suggestion
        return f"amazonaws.com.cn"
```